### PR TITLE
solution

### PR DESCRIPTION
**Pull request by illiashcherbak**

### CODE REVIEW

File: src/sortStudents.ts [hunk 1]
```diff
@@ -1,16 +1,74 @@
...
+    case SortType.Name:
+    case SortType.Surname:
+      sortedStudentsList = sortedStudentsList
+        .sort((currStudent, nextStudent) => {
+          return order === 'asc'
+            ? currStudent[sortBy].localeCompare(nextStudent[sortBy])
+            : nextStudent[sortBy].localeCompare(currStudent[sortBy]);
+        });
+
+      break;
```
Comment (anastasiia-bilska) [inline src/sortStudents.ts:44]:
```suggestion
  const sortedStudentsList = [...students];

  switch (sortBy) {
    case SortType.Name:
    case SortType.Surname:
      return sortedStudentsList
        .sort((currStudent, nextStudent) => {
          return order === 'asc'
            ? currStudent[sortBy].localeCompare(nextStudent[sortBy])
            : nextStudent[sortBy].localeCompare(currStudent[sortBy]);
        });
```

Comment (anastasiia-bilska) [reply src/sortStudents.ts:44]: sort is a mutable, you don't need to assign the result and you can use const for sortedStudentsList. You also can return sorted list, so you don't need to write 'break'. Same for next cases

File: src/sortStudents.ts [hunk 2]
```diff
@@ -1,16 +1,74 @@
...
+
+    default:
+      break;
+  }
+
+  return sortedStudentsList;
 }
```
Comment (anastasiia-bilska) [inline src/sortStudents.ts:74]:
```suggestion
    default:
       return sortedStudentsList;  
    }
}
```

**Review by anastasiia-bilska — CHANGES_REQUESTED** [review]
Good! Please, check my comments

**Review by anastasiia-bilska — APPROVED** [review]
Well done!

File: src/sortStudents.ts [hunk 3]
```diff
@@ -1,16 +1,66 @@
...
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverageGrades(gradesArray: number[]): number {
+  const sumOfGrades = gradesArray.reduce((acc, curr) => acc + curr, 0);
+  const avarageAge = sumOfGrades / gradesArray.length;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return avarageAge;
```
Comment (solaryasha) [inline src/sortStudents.ts:24]:
```suggestion
  return sumOfGrades / gradesArray.length;;
```

File: src/sortStudents.ts [hunk 4]
```diff
@@ -1,16 +1,66 @@
...
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedStudentsList = [...students];
```
Comment (solaryasha) [inline src/sortStudents.ts:32]:
```suggestion
  const copiedStudents = [...students];
```

**Review by solaryasha — APPROVED** [review]

**Pull request closed without merging**